### PR TITLE
feat: use next/script in ConsentManager

### PR DESCRIPTION
**Pull request by BRKalow**
🎟️ [Asana Task]()
🔍 [Preview Link](https://react-components-git-{branch-slug}-hashicorp.vercel.app)

---

## Description

Refactoring ConsentManager to use `next/script` under the hood, as well as some refactoring to a functional component and hooks.

* Adds a new `strategy` attribute on services, which we can specify to pass along a desired loading strategy to `next/script`
* Refactors the top-level `ConsentManager` component to TS and away from a class component
* Cleans up tests and internal logic

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/consent-manager/index.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        />
+      )}
+      <SegmentScript
+        preferences={preferences}
+        writeKey={props.segmentWriteKey}
+      />
+      <CustomScripts
+        preferences={preferences}
+        services={props.additionalServices}
+      />
```
Comment (BRKalow) [inline packages/consent-manager/index.tsx:120]: This replaces the previous `loadAnalytics` method. It's a bit more declarative and easier to trace now, these components will inject the necessary scripts based on the user's preferences.

File: packages/consent-manager/loader/index.tsx [hunk 2]
```diff
@@ -50,10 +57,10 @@ export default function createConsentManager({
   }
 
   // next we build the config objct, kicking it off with the default values
-  let config: ConsentManagerConfig = {
-    version: 4,
-    companyName: 'HashiCorp',
-    privacyPolicyLink: 'https://www.hashicorp.com/privacy',
+  let config: ConsentManagerProps = {
```
Comment (BRKalow) [inline packages/consent-manager/loader/index.tsx:60]: Standardizing on shared types

**Review by mwickett — APPROVED** [review]
This looks really nice. Great to see an older component like this get cleaned up, and get improvements.

I've given the code a high-level look, and tested in the Swingset preview to see that scripts are in fact being loaded when consent is given.

**Review by jescalan — COMMENTED** [review]
Looks great!

**Review by jescalan — APPROVED** [review]

**Review by thiskevinwang — APPROVED** [review]
LGTM!

Nit: Maybe next >=11 should be specified as a peer dep? ...since `next/script` is an [11+](https://nextjs.org/blog/next-11) API

Which also makes me curious: Is this a major or minor change?

**Comment by BRKalow** [comment]
@thiskevinwang Yeah good call I'll add next to peer deps. I think strictly by the book this could be a major for the reason you said, but since all of our stuff is on next 11 and we control the apps this runs in it is reasonable for it to be a minor.

**Pull request merged**